src/nomad_measurements/xrd/schema.py

```diff
@@ -1000,7 +970,9 @@ def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger'):
         super().normalize(archive, logger)
         if not self.results:
             return
-        self.write_nx_section_and_create_file(archive, logger)
+        write_nx_section_and_create_file(
+            archive, logger, self.generate_nexus_file, nxs_as_entry=True
```

Comment:
`self.generate_nexus_file` does not seem to work as intended. Even if it's set as false, the nexus file is still generated

Reply:
Sorry it is a wrong logic. I fix it. Can you try it.

Reply:
looks good! can you update the the argument descriptions in docstring of `write_nx_section_and_create_file`?